src/accelerate/utils/other.py

```diff
@@ -207,6 +213,42 @@ def save(obj, f, save_on_each_node: bool = False, safe_serialization: bool = Fal
         save_func(obj, f)
 
 
+# The following are considered "safe" globals to reconstruct various types of objects when using `weights_only=True`
+# These should be added and then removed after loading in the file
+TORCH_SAFE_GLOBALS = [
+    # numpy arrays are just numbers, not objects, so we can reconstruct them safely
+    np.core.multiarray._reconstruct,
+    np.ndarray,
+]
+
+
+def load(f, map_location=None, **kwargs):
+    """
+    Compatible drop-in replacement of `torch.load()` which allows for `weights_only` to be used if `torch` version is
+    2.0.0 or higher. Otherwise will ignore the kwarg.
+
+    Will also add (and then remove) an exception for numpy arrays
+
+    Args:
+        f:
+            The file (or file-like object) to use to load the data
+        map_location:
+            a function, `torch.device`, string or a dict specifying how to remap storage locations
+        **kwargs:
+            Additional keyword arguments to pass to `torch.load()`.
+    """
+    if is_weights_only_available():
+        if "weights_only" not in kwargs:
+            kwargs["weights_only"] = True
+        torch.serialization.add_safe_globals(TORCH_SAFE_GLOBALS)
+    else:
+        kwargs.pop("weights_only", None)
+    loaded_obj = torch.load(f, map_location=map_location, **kwargs)
+    if is_weights_only_available():
+        torch.serialization.clear_safe_globals()
```

Comment:
Probably better to put this all in a `try ... finally` to ensure that the safe globals are cleared at the end.

Reply:
Yes, sorry about this, there is a `safe_globals` context manager, but that is only available in nightlies at the moment :/ https://pytorch.org/docs/main/notes/serialization.html#torch.serialization.safe_globals